src/cdk/a11y/focus-monitor/focus-monitor.spec.ts

```diff
@@ -393,6 +395,35 @@ describe('cdkMonitorFocus', () => {
   });
 });
 
+describe('FocusMonitor observable stream', () => {
+  let fixture: ComponentFixture<MonitoredElementRequiringChangeDetection>;
+  let buttonElement: HTMLElement;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [A11yModule],
+      declarations: [
+        MonitoredElementRequiringChangeDetection,
+      ],
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(MonitoredElementRequiringChangeDetection);
+    fixture.detectChanges();
+    buttonElement = fixture.componentInstance.button.nativeElement;
+    patchElementFocus(buttonElement);
+  });
+
+  it('should emit inside the NgZone', fakeAsync(() => {
+    fixture.detectChanges();
+    expect(buttonElement.innerText).toBe('');
+    buttonElement.focus();
+    fixture.detectChanges();
+    expect(buttonElement.innerText).toBe('program');
```

Comment:
Rather than having the test go through change detection, you should be able to assert the same using `NgZone`:

```ts
const spy = jasmine.createSpy('zone spy');

focusMonitor.monitor(element).subscribe(() => {
  spy(NgZone.isInAngularZone());
});

expect(spy).toHaveBeenCalledWith(true);
```

Reply:
neat, didn't know that method existed